app/src/lib/stores/app-store.ts

```diff
@@ -2294,7 +2293,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
   private async matchGitHubRepository(
     repository: Repository
   ): Promise<IMatchedGitHubRepository | null> {
-    const remote = await getDefaultRemote(repository)
+    const gitStore = this.getGitStore(repository)
+    const remote = gitStore.defaultRemote
     return remote ? matchGitHubRepository(this.accounts, remote.url) : null
```

Comment:
`return remote !== null ? matchGitHubRepository(this.accounts, remote.url) : null`
I know you didn't touch this, but it's so close to your changes that I think this would be a great cleanup.